src/NuGet.Core/NuGet.Packaging/ContentModel/ContentItemCollection.cs

```diff
@@ -358,69 +391,155 @@ private static bool IsValidAsset(string path)
             return false;
         }
 
-        private class GroupComparer : IEqualityComparer<ContentItem>
+        internal class GroupComparer : IEqualityComparer<ContentItem>
         {
             public static readonly GroupComparer DefaultComparer = new GroupComparer();
 
             public int GetHashCode(ContentItem obj)
             {
                 var hashCode = 0;
-                foreach (var property in obj.Properties)
+                if (obj._properties != null)
                 {
-                    if (property.Key.Equals("tfm_raw", StringComparison.Ordinal))
+                    foreach (var property in obj._properties)
                     {
-                        // We store the raw version of the TFM, but we don't want it to affect the result.
-                        continue;
-                    }
 #if NETFRAMEWORK || NETSTANDARD
-                    hashCode ^= property.Key.GetHashCode();
+                        hashCode ^= property.Key.GetHashCode();
```

Comment:
I know nothing about good hash code design. But our own `HashCodeCombiner`, and also [.NET's HashCode](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/HashCode.cs) both do multiplcations and bitshifts.

Should we port this to use HashCodeCombiner?

Reply:
We could potentially yeah.

In the interest of getting this PR merged, I can do that in a follow-up.